apis/v1alpha1/shared_types.go

```diff
@@ -145,8 +147,7 @@ type AdminNetworkPolicyIngressPeer struct {
 }
 
 // CIDR is an IP address range in CIDR notation (for example, "10.0.0.0/8" or "fd00::/8").
-// This string must be validated by implementations using net.ParseCIDR
-// TODO: Introduce CEL CIDR validation regex isCIDR() in Kube 1.31 when it is available.
+// +kubebuilder:validation:XValidation:rule="isCIDR(self)",message="Invalid CIDR format provided"
 // +kubebuilder:validation:XValidation:rule="self.contains(':') != self.contains('.')",message="CIDR must be either an IPv4 or IPv6 address. IPv4 address embedded in IPv6 addresses are not supported"
```

Comment:
should remove the old CEL too right?

Reply:
yea I had the same doubt, I think isCIDR also covers the old CEL so I shouldn't need it..., @JoelSpeed can you double confirm so that I can remove it ?

Reply:
Yep, I think it's safe to remove. I'm pretty sure we don't allow embedded addresses in the CEL rule.

I'd advise making sure you have a test for an embedded address just to make sure though

Reply:
ack I'll remove this and also I have a tracker for adding tests: https://github.com/kubernetes-sigs/network-policy-api/issues/148 I'll get on that as well

Reply:
removed.

Reply:
removed.